src/app/pages/quizzes-catalog/quizzes-catalog.component.ts

```diff
@@ -16,21 +16,35 @@ import { StoreService } from '../../services/store.service';
   templateUrl: './quizzes-catalog.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizzesCatalogComponent implements OnInit {
+export class QuizzesCatalogComponent implements OnInit, OnDestroy {
   categories$!: Observable<CategoryModel[]>;
   isLoading$!: Observable<boolean>;
+  subscription!: Subscription;
   
   constructor(private storeService: StoreService) { }
-
+  
   ngOnInit(): void {
+    this.subscription = this.storeService.getCategories()
+    .pipe(take(1))
+    .subscribe((categories) => {
+      if (!categories || categories.length === 0) {
```

Comment:
Instead of this logic we can make that loadCategories() will return current store categories if they are present, else - load the api